src/node_sqlite.cc

```diff
@@ -67,6 +67,63 @@ using v8::Value;
+    from, unreachable_message, isolate, use_big_int_args, ...)                 \
+  switch (sqlite3_##from##_type(__VA_ARGS__)) {                                \
+    case SQLITE_INTEGER: {                                                     \
+      sqlite3_int64 val = sqlite3_##from##_int64(__VA_ARGS__);                 \
+      if (use_big_int_args) {                                                  \
+        return BigInt::New(isolate, val);                                      \
+      } else if (std::abs(val) <= kMaxSafeJsInteger) {                         \
+        return Number::New(isolate, val);                                      \
+      } else {                                                                 \
+        return MaybeLocal<Value>();                                            \
+      }                                                                        \
+      break;                                                                   \
+    }                                                                          \
+    case SQLITE_FLOAT: {                                                       \
+      return Number::New(isolate, sqlite3_##from##_double(__VA_ARGS__));       \
+      break;                                                                   \
+    }                                                                          \
+    case SQLITE_TEXT: {                                                        \
+      const char* v =                                                          \
+          reinterpret_cast<const char*>(sqlite3_##from##_text(__VA_ARGS__));   \
+      return String::NewFromUtf8(isolate, v).As<Value>();                      \
+    }                                                                          \
+    case SQLITE_NULL:                                                          \
+      return Null(isolate);                                                    \
+    case SQLITE_BLOB: {                                                        \
+      size_t size = static_cast<size_t>(sqlite3_##from##_bytes(__VA_ARGS__));  \
+      auto data = reinterpret_cast<const uint8_t*>(                            \
+          sqlite3_##from##_blob(__VA_ARGS__));                                 \
+      auto store = ArrayBuffer::NewBackingStore(isolate, size);                \
+      memcpy(store->Data(), data, size);                                       \
+      auto ab = ArrayBuffer::New(isolate, std::move(store));                   \
+      return Uint8Array::New(ab, 0, size);                                     \
+    }                                                                          \
+    default:                                                                   \
+      UNREACHABLE(unreachable_message);                                        \
+  }
+
+MaybeLocal<Value> SQLiteValueToJS(Isolate* isolate,
```

Comment:
Can we have a function or a macro instead of both?

Reply:
Done